src/internal/lokiutil/client/client.go

```diff
@@ -71,8 +73,15 @@ func (c *Client) doRequest(ctx context.Context, path, query string, quiet bool,
 	if err != nil {
 		return errors.EnsureStack(err)
 	}
-
 	defer resp.Body.Close()
+
+	if resp.StatusCode >= http.StatusBadRequest {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			body = append(body, []byte(fmt.Sprintf("\nreading body: %v", err))...)
+		}
+		return errors.EnsureStack(errors.Errorf("error response from loki: %v (body: %s)", resp.Status, body))
+	}
 	return errors.EnsureStack(json.NewDecoder(resp.Body).Decode(out))
```

Comment:
Rather than printing that the body contained `"\nreading body: …"` when it did not, why not this instead:

```go
	if resp.StatusCode >= http.StatusBadRequest {
		body, err := io.ReadAll(resp.Body)
		if err != nil {
			return errors.EnsureStack(errors.Errorf("error response from loki: %v (body: %q; err: %v)", resp.Status, body, err))
		}
		return errors.EnsureStack(errors.Errorf("error response from loki: %v (body: %q)", resp.Status, body))
	}
```

FWIW, I also quoted the body string, which might help future parsing.

Reply:
Done.